### ISSUE TITLE
IN condition on non-list

### ISSUE THREAD

**Issue by Zhouenze** [partiql/partiql-lang-kotlin#362]
Hi there,

I noticed that if IN condition has non-list parameter, PartiQL returns false instead of throwing Exception. I found this confusing  because people may forget to add `()` around parameter but don't get error notification. As a result, they may end up executing wrong statement for a long time.

Meanwhile, returning false is also desirable in some cases. Like when we have 2 documents in `{id:1, inside: string}, {id:1, inside: [1]}` and execute `SELECT * FROM tb WHERE id IN inside` (inside here refers to corresponding field in same document). If IN accepts non-list but consider it false, above statement could succeed and return `{id:1, inside: [1]}`. But if IN throws on non-list, then above statement simply fails.

To sum up, I wonder if there could be a way to notify customer of possibly wrong type in IN condition, but don't have specific idea on how to do it (maybe shouldn't take action at all). Just want to raise attention on this scenario. Or maybe a bigger request is: For queries that are valid PartiQL but probably not what user intended (write by mistake), could we provide some form of suggestion or notification?

Thanks!
